File: from-1801-to-1900/1838-frequency-of-the-most-frequent-element/solution.py
class Solution:
    def maxFrequency(self, nums: List[int], k: int) -> int:
        nums.sort()
        left, right = 0, 0
        max_width = 0
        total = 0

        while right < len(nums):
            total += nums[right]
            
            while nums[right] * (right - left + 1) > total + k:
                total -= nums[left]
                left += 1

            max_width = max(max_width, right - left + 1)
            right += 1

        return max_width

File: from-1801-to-1900/1838-frequency-of-the-most-frequent-element/solution.go
func maxFrequency(nums []int, k int) int {
    sort.Ints(nums)

    left := 0
    result, sum := 1, 0

    for right := 0; right < len(nums); right ++ {
        sum += nums[right]
        if nums[right] * (right - left + 1) > sum + k {
            sum -= nums[left]
            left += 1
        }
        if right - left + 1 > result {
            result = right - left + 1
        }
    }
    return result
}